tracer/src/Datadog.Trace.Tools.dd_dotnet/CreatedumpCommand.cs

```diff
@@ -0,0 +1,676 @@
+         || assemblyName.StartsWith("nunit.framework,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("OpenTelemetry,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("OpenTelemetry.Api,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("Oracle.DataAccess,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("Oracle.ManagedDataAccess,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("RabbitMQ.Client,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("Serilog,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("ServiceStack.Redis,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("StackExchange.Redis,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("StackExchange.Redis.StrongName,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("System,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("xunit.execution.desktop,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("xunit.execution.dotnet,", StringComparison.Ordinal)
+         || assemblyName.StartsWith("Yarp.ReverseProxy,", StringComparison.Ordinal))
+        {
+            return false;
+        }
+
+        return true;
+    }
+
+    private static bool IsTelemetryEnabled()
+    {
+        var value = Environment.GetEnvironmentVariable(ConfigurationKeys.Telemetry.Enabled);
+
+        if (string.Equals(value, "false", StringComparison.OrdinalIgnoreCase) || value == "0")
+        {
+            return false;
+        }
+
+        return true;
+    }
+
+    private static bool IsMethodSuspicious(ClrMethod method)
+    {
+        var assemblyName = Path.GetFileName(method.Type?.Module?.Name ?? string.Empty);
+
+        if (assemblyName.StartsWith("Datadog", StringComparison.OrdinalIgnoreCase))
+        {
+            var typeName = method.Type!.Name;
+
+            if (typeName != null)
+            {
+                if (typeName == "Datadog.Trace.ClrProfiler.AutoInstrumentation.AspNetCore.BlockingMiddleware")
+                {
+                    return false;
+                }
+
+                // Datadog.Trace.ClrProfiler.CallTarget.Handlers.Continuations.TaskContinuationGenerator`4.SyncCallbackHandler.<ContinuationAction>d__4.MoveNext()
+                if (typeName.Contains("TaskContinuationGenerator")
+                 && typeName.Contains("ContinuationAction")
+                 && method.Name == "MoveNext")
+                {
+                    // Very likely to be an async continuation
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        if (method.Type!.Module!.IsDynamic && assemblyName.StartsWith("DuckType"))
+        {
+            return true;
+        }
+
+        return false;
+    }
+
+    private static bool ParseArguments(string[] arguments, out int pid, out int? signal, out int? crashThread)
```

Comment:
This seems like a great candidate for making internal and throwing a bunch of test cases at it in a unit test, especially when you say things like "it can get tricky" 😉

Reply:
You're right, I forgot the unit tests